### PR TITLE
ensure HTTPSconnection.is_verified is set properly

### LINKED ISSUE #3267: https_proxy to http_host leads to `is_verfied==True`
(followup to #3130)

### Subject

@illia-v , the test you provided within https://github.com/urllib3/urllib3/pull/3149#pullrequestreview-1683158046

```py
     def test_https_proxy(self) -> None:                                                                                                                                                                                                                                                                                                                                                                            
         with proxy_from_url(self.https_proxy_url, ca_certs=DEFAULT_CA) as https:                                                                                                                                                                                                                                                                                                                                   
[...]                                                                                                                                                                                                                                                                                                                                                                                                     
             r = https.request("GET", f"{self.http_url}/")                                                                                                                                                                                                                                                                                                                                                          
             assert r.status == 200                                                                                                                                                                                                                                                                                                                                                                                 
+            pool = list(https.pools._container.values())[-1]                                                                                                                                                                                                                                                                                                                                                       
+            connection = pool.pool.queue[-1]                                                                                                                                                                                                                                                                                                                                                                       
+            assert connection.is_verified is False                                                                                                                                                                                                                                                                                                                                                                 
+            assert connection.proxy_is_verified is True                                                                                                                                                                                                                                                                                                                                                          
                                                             
```

Thus, the 

* `proxy_is_verifed is True`, `is_verified is False`

I tend to agree with this logic, but the current code does

* `proxy_is_verifed is True`, `is_verified is True`.

This seems like an additional (besides #3130) bug (or at least flaw).

You can review the behavior within 

- https://github.com/urllib3/urllib3/pull/3266

the relevant test is `test_is_verified_https_proxy_to_http_target`

### PR DESCRIPTION
**Pull request by abebeos**
- closes #3267

### Followup Issue

Relevant code needs to be revised & (self)documented (to avoid such kind of bugs in future)

### CODE REVIEW

File: src/urllib3/connection.py [hunk 1]
```diff
@@ -663,11 +670,11 @@ def connect(self) -> None:
...
-        # TODO: Set correct `self.is_verified` in case of HTTPS proxy +
-        #       HTTP destination, see
-        #       `test_is_verified_https_proxy_to_http_target` and
-        #       https://github.com/urllib3/urllib3/issues/3267.
-        self.is_verified = sock_and_verified.is_verified
+        # https://github.com/urllib3/urllib3/issues/3267.
+        if self.proxy_is_forwarding:
+            self.is_verified = False
+        else:
+            self.is_verified = sock_and_verified.is_verified
```
Comment (sigmavirus24) [inline src/urllib3/connection.py:677]: Why not more simply

```
self.is_verified = sock_and_verified.is_verified and not self.proxy_is_forwarding
```

**Review by sigmavirus24 — CHANGES_REQUESTED** [review]

File: src/urllib3/connection.py (hunk 1 above)
Comment (abebeos) [reply src/urllib3/connection.py:677]: Personal preference.

File: src/urllib3/connection.py [hunk 2]
```diff
@@ -265,6 +265,13 @@ def is_connected(self) -> bool:
     def has_connected_to_proxy(self) -> bool:
         return self._has_connected_to_proxy
 
+    @property
+    def proxy_is_forwarding(self) -> bool:
```
Comment (sethmlarson) [inline src/urllib3/connection.py:269]: Can you open an issue for creating a property for `proxy_is_tunnel`?

**Review by sethmlarson — APPROVED** [review]
LGTM

**Pull request merged**